tests/draft2019-09/ref.json

```diff
@@ -355,5 +363,20 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "use of ref that doesn't exist",
+        "schema": {
+            "properties": {
+                "bar": {"$ref": "#/properties/not-found"}
```

Comment:
Correct me if I'm wrong, since yeah clearly not yet an expert on 2019-09, but I don't think this one is correct for say, older specs -- in older specs the result here isn't an invalid instance, it's undefined, or an error entirely during ref resolution.

Do you know that that's changed in 2019-09? (Will verify it myself as well).

Reply:
I'll back this out. Backstory is that with draft 2019-09 and `recursiveRef` it's no longer possible to statically resolve & verify references. It raises an interesting question of how one should report dereference errors which, to my knowledge, is unspecified by the standard.

I opted to report them as an error annotation, the same as any other validation error, which allows validation to complete with an "invalid" result. Others might reasonably make a different decision.

Reply:
Yeah, mine raises an immediate exception.

Reply:
@jgraettinger the `$recursiveRef` changes shouldn't impact `$ref`.  Since the commit is gone I can't see the larger context of this comment so maybe I'm missing something?

`$ref` is entirely possible to statically resolve without an instance.  Also, the initial URI value of `$recursiveRef` can be resolved against its base statically (you just don't know if it's going to get adjusted until runtime).

Let me know if I'm off-track here, but it's an important property of `$ref` that it can be resolved without an instance and I want to make sure there's no problem with that.

Reply:
The point was/is "referencing a location that doesn't exist is neither valid nor invalid as a result of validation, it's just an error, and implementation defined as to how that's reported", at least as far as I know previously.

That's still true right?